Replace Google TM code with Matomo TM code

katleonard opened the pull request:
Replacing Google Tag Manage code with Matomo Tag Manager code

katleonard committed:
Replace Google TM code with Matomo TM code

Replacing Google Tag Manage code with Matomo Tag Manager code

HaSistrunk commented:
Ok! I see a few other things:

1. I'm realizing that we don't need the "if statement" in lines 2 and 12 at all (the `if jekyll.environment == "production"` stuff). This is a conditional statement that prevents the Matomo script from running unless the site is deployed on the live production environment. However, in our setup we are actually controlling that in the Matomo settings for the website instead of here in the site code. So you can delete those two lines!

2. The Matomo script you used doesn't look like it's the one that corresponds to the Library Site container in Matomo. Each site in Matomo Tag Manager has a unique container id, so it's best to copy the script directly from the "Install Code" script that Matomo provides for each site that you can find in Tag Manager on the left-side menu at the bottom.

3. There is a Google Tag Manager (noscript) block of code that also needs to be deleted in a different file (`_layouts/default.html) lines 6-9. We don't need to replace that with anything - it's an extra bit of code that Google requires and Matomo does not.

katleonard committed:
Updating Head Matomo TM code from GTM Part 2

Took out the 'if statements' from line 2 and line 12 'the if jekyll.environment == "production' as requested. Replaced the Matomo code with the dev Matomo code from Library Discovery Matomo Tag Manager page- clicking on Install code on that page and selecting dev from the drop down menu at the top of the page.  Copy and pasted that code here.

katleonard committed:
Updating default removing GTM noscript

Updating default by removing Google Tag Manager (noscript) block of code that relates to Google and Matomo doesn't require.

HaSistrunk commented on _includes/head.html:
```diff
@@ -1,14 +1,12 @@
+var _mtm = window._mtm = window._mtm || [];
+_mtm.push({'mtm.startTime': (new Date().getTime()), 'event': 'mtm.Start'});
+var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
+g.async=true; g.src='https://cdn.matomo.cloud/rockarch.matomo.cloud/container_SdzKujX7_dev_cc9545323ac9afaa195e6699.js'; s.parentNode.insertBefore(g,s);
```
```suggestion
g.async=true; g.src='https://cdn.matomo.cloud/rockarch.matomo.cloud/container_SdzKujX7.js'; s.parentNode.insertBefore(g,s);
```

HaSistrunk replied:
We can just use the live container environment. Happy to talk through our approach in the next sprint meeting.

HaSistrunk committed:
Merge changes from 'development' for build to pass

katleonard committed:
MTM code updated to Live not dev

Matomo code updated not using dev but live code to install

Co-authored-by: Hannah Sistrunk <hsistrunk@gmail.com>

daremiyo reviewed (approved):
Looks good with the changes you've made!
